rtt_tf/rtt_tf-component.cpp

```diff
@@ -196,6 +205,29 @@ namespace rtt_tf
         }
       }
     }
+    // Publish the geometry messages obtained from the lookupTransform on
+    // tracked transformations
+    // Range for not supported in ISO C++ 11
+    // for (const auto& tracker : ports_trackers) {
+    //   auto msg = lookupTransform(tracker.first.first, tracker.first.second);
+    //   tracker.second->write(msg);
+    // }
+    for(std::map<std::pair<std::string, std::string>, OutputPortGeometryTransfromStampedPtr>::iterator
+      it_tracker = ports_trackers.begin(); it_tracker != ports_trackers.end(); ++it_tracker) {
+      geometry_msgs::TransformStamped msg = lookupTransform(it_tracker->first.first, it_tracker->first.second);
+      it_tracker->second->write(msg);
+     }
+  }
+
+  bool RTT_TF::startHook()
+  {
+    if (0 == getPeriod()) {
+      RTT::log(RTT::Warning) << "The period of the component is 0 (zero), so no"
+        " updates form TF will be published automatically" << RTT::endlog();
+    }
+    buffer_core = BufferCorePtr(new tf2::BufferCore());
```

Comment:
Initializing by calling `buffer_core.reset(new tf2::BufferCore())` instead is preferred.